src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManager.cs

```diff
@@ -103,6 +116,82 @@ public override async Task SendConnectionAsync(string connectionId, string metho
             }
         }
 
+#if NET7_0_OR_GREATER
+        public override async Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object[] args, CancellationToken cancellationToken = default)
+        {
+            if (IsInvalidArgument(connectionId))
+            {
+                throw new ArgumentNullException(nameof(connectionId));
+            }
+
+            if (IsInvalidArgument(methodName))
+            {
+                throw new ArgumentNullException(nameof(methodName));
+            }
+
+            var invocationId = _clientInvocationManager.Caller.GenerateInvocationId(connectionId);
+
+            // Exception handling follows https://source.dot.net/#Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs,349
+            try
+            {
+                var message = AppendMessageTracingId(new ClientInvocationMessage(invocationId, connectionId, _callerId, SerializeAllProtocols(methodName, args, invocationId)));
+                await WriteAsync(message);
+                if (_clientConnectionManager.ClientConnections.TryGetValue(connectionId, out var clientConnectionContext))
+                {
+                    var instanceId = clientConnectionContext.InstanceId;
+                    var task = _clientInvocationManager.Caller.AddInvocation<T>(connectionId, invocationId, instanceId, cancellationToken);
+                    return await task;
+                }
+            }
+            catch
+            {
+                // Use `TryCompleteResult` to remove the failed invoation.
+                // If the invocation was not added by caller, it will be ignored.
+                // The content of error message is useless and will not be exposed to client.
+                _clientInvocationManager.Caller.TryCompleteResult(connectionId, CompletionMessage.WithError(invocationId, "Invocation Failed"));
+                throw;
```

Comment:
will exception thrown be logged?

Reply:
will be logged. I add an extra log to record richer information for user

Reply:
why still throw?